site/content/entry/citra-progress-report-2017-q4.md

```diff
@@ -0,0 +1,202 @@
+streamed in, which is used extensively across 3DS apps, such as the eShop streaming
+downloads directly into `am` so that it installs at the same time, or `dlp` using
+it to stream Download Play CIAs from one 3DS to another directly.
+
+The CIA format starts off with a header that describes where each component of
+itself starts within the stream and a bitfield. This is then followed by a
+certificate signed by Nintendo, the CIA's ticket, and the title metadata (or TMD
+for short). The TMD then contains a list of every CXI file the CIA could possibly
+have, along with their name, size, etc.
+
+{{< figure src="/images/entry/citra-progress-report-2017-q4/oot3dcia.png" 
+    title="A diagram of *The Legend
+of Zelda: Ocarina of Time 3D*'s CIA" >}}
+
+The astute reader might've noticed I said that the TMD contains a list of every
+*possible* CXI. This is because a CIA might not contain every CXI for the game,
+such as in the case of DLC. The bitfield I mentioned in the header is used to list
+off which of the entries in the TMD actually exists within that CIA. This is
+(ab)used in some cases, such as the Home Menu Themes, in that every single theme
+that exists is actually a different CXI within the same CIA. Just tick the themes
+the user owns in the bitfield and then attach those CXIs at the end of it. Dead
+simple way for it to work perfectly with both streaming of the file, and making
+personalized CIAs for each eShop user.
+
+{{< figure src="/images/entry/citra-progress-report-2017-q4/menucia.png" 
+    title="A tiny portion of the Home Menu Themes CIA" >}}
+
+For the `am` side of streaming installations, the way it handles a request to
+install an application is by creating and then giving the requester a handle to
+a virtual file. This leaves most of the busy work to `am`, while the app only needs
+to worry about writing the CIA file into that handle.
+
+[shinyquagsire23](https://github.com/shinyquagsire23) reimplemented the entirety
+of how Citra handles NCCH files, which quickly paved the way to implementing `am`
+in Citra. This allowed users to use FBI to install  applications from its virtual
+SD card. He and [BreadFish64](https://github.com/BreadFish64) also went a step
+further and added support for installing CIAs via the SDL and Qt frontends
+directly, removing the need to use FBI. Do note that you cannot yet run installed
+applications and games directly, but CXI executables can access the virtual SD
+card for DLC, updates, etc.
+
+## [Qtifw build installer](https://github.com/citra-emu/citra/pull/2966) by [j-selby](https://github.com/j-selby)
+
+Citra now has a fancy new installer and updater, thanks to [j-selby](https://github.com/j-selby)
+and [jroweboy](https://github.com/jroweboy)'s efforts! If you haven't already, we
+strongly recommend you download and install Citra through it, since it will make
+updating as easy as re-running the installer. You can check it out over [here](/download/),
+
```

Comment:
Maybe include a blurb about the "Check for update" feature in the help menu

Reply:
I didn't realize there was an update feature from within Citra, since every build I've checked (including the one I compiled from `master` a few hours ago) doesn't have a "Check for Update" button in the Help menu. Is it only enabled with a CMake switch?

Reply:
@anodium I believe only builds from the installer have the "Check for Updates" button.